OSVR-Unity/Assets/OSVRUnity/src/OsvrRenderManager.cs

```diff
@@ -0,0 +1,254 @@
+
+using UnityEngine;
+using System.Collections;
+using System.Runtime.InteropServices;
+using System.Text.RegularExpressions;
+using System;
+
+namespace OSVR
+{
+    namespace Unity
+    {
+        //*This class is a wrapper for the OSVR-Unity Rendering Plugin osvrUnityRenderingPlugin.dll,
+        // which brings in functionality from the OSVR RenderManager project. RenderManager features inculde:
+        // - DirectMode -- compatible with nVidia cards with a driver that has been modified to white-list the display that you are using.
+        // - TimeWarp
+        // - Distortion Correction
+        //
+        // osvrUnityRenderingPlugin.dll, osvrRenderManager.dll, SDL2.dll, and glew32.dll must be in the Plugins/x86 or x64 folders.
+        // Requires Unity 5.2+
+        //*/
+        public class OsvrRenderManager : MonoBehaviour
+        {
+            public const int RENDER_EVENT = 0;
+            public const int SHUTDOWN_EVENT = 1;
+            public const int UPDATE_RENDERINFO_EVENT = 2;
+            private const string PluginName = "osvrUnityRenderingPlugin";
+
+            // Allow for calling into the debug console from C++
+            [DllImport(PluginName)]
+            private static extern void LinkDebug([MarshalAs(UnmanagedType.FunctionPtr)]IntPtr debugCal);
+            [UnmanagedFunctionPointer(CallingConvention.StdCall)]
+            private delegate void DebugLog(string log);
+            private static readonly DebugLog debugLog = DebugWrapper;
+            private static readonly IntPtr functionPointer = Marshal.GetFunctionPointerForDelegate(debugLog);
+            private static void DebugWrapper(string log) { Debug.Log(log); }
+
+            //get the render event function that we'll call every frame via GL.IssuePluginEvent
+            [DllImport(PluginName, CallingConvention = CallingConvention.StdCall)]
+            private static extern IntPtr GetRenderEventFunc();
+
+            //Pass a pointer to a texture (RenderTexture.GetNativeTexturePtr()) to the plugin
+            [DllImport(PluginName, CallingConvention = CallingConvention.Cdecl)]
+            private static extern void SetColorBufferFromUnity(System.IntPtr texturePtr, int eye);
+
+            //Create a RenderManager object in the plugin, passing in a ClientContext
+            [DllImport(PluginName, CallingConvention = CallingConvention.Cdecl)]
+            private static extern Byte CreateRenderManagerFromUnity(OSVR.ClientKit.SafeClientContextHandle /*OSVR_ClientContext*/ ctx);
+
+            [StructLayout(LayoutKind.Sequential)]
+            public struct OSVR_ViewportDescription
+            {
+                public double left;    //< Left side of the viewport in pixels
+                public double lower;   //< First pixel in the viewport at the bottom.
+                public double width;   //< Last pixel in the viewport at the top
+                public double height;   //< Last pixel on the right of the viewport in pixels
+            }
```

Comment:
I think the fix is to get RenderManager and Core to use the same structure, so will file a separate issue.  Ryan and I can fight about which way the structure is.